rust-runtime/aws-smithy-http/src/event_stream/sender.rs

```diff
@@ -77,16 +117,12 @@ where
             signer: Box::new(signer),
             stream,
             end_signal_sent: false,
-            _phantom: Default::default(),
         }
     }
 }
 
-impl<T, E> Stream for MessageStreamAdapter<T, E>
-where
-    E: StdError + Send + Sync + 'static,
-{
-    type Item = Result<Bytes, SdkError<E>>;
+impl<T> Stream for MessageStreamAdapter<T> {
+    type Item = Result<Bytes, SdkError<MessageStreamError>>;
```

Comment:
The `SdkError` here needs to be generic over the modeled error type so that customers can match on the streaming errors. I don't think we need to have `MessageStreamError` since `SdkError` already encompasses its functionality.